src/core/lib/surface/server.cc

```diff
@@ -1180,14 +1180,7 @@ void grpc_server_setup_transport(
     chand->channelz_socket_uuid = 0;
   }
 
-  size_t cq_idx;
-  for (cq_idx = 0; cq_idx < s->cq_count; cq_idx++) {
-    if (grpc_cq_pollset(s->cqs[cq_idx]) == accepting_pollset) break;
-  }
-  if (cq_idx == s->cq_count) {
-    /* completion queue not found: pick a random one to publish new calls to */
-    cq_idx = static_cast<size_t>(rand()) % s->cq_count;
-  }
+  size_t cq_idx = static_cast<size_t>(rand()) % s->cq_count;
```

Comment:
can you think of something that's cheaper that this? both rand and "%" are expensive.

Perhaps hash the transport pointer instead?

Reply:
Hashing the transport pointer is easy enough to do. I don't see a way around % though, unless we force power of two and mask.

Which, actually, is what I'll do. We'll shard so the number of shards is the next power of two greater than or equal to the number of processors, gpr_murmur_hash the transport and bitmask to get the right shard.

Reply:
done, PTAL